### PR TITLE
Up groups

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
required tools to fix rounding issues (due to exchange max precision) on multiple sell profits (ex: position of 20.1 translated into 2 sell orders of 10 instead of one of 10.1 and one of 10)

### CODE REVIEW

File: tests/personal_data/orders/test_decimal_order_adapter.py [hunk 1]
```diff
@@ -57,6 +57,22 @@ async def test_decimal_adapt_price():
     assert personal_data.decimal_adapt_price(symbol_market,
                                              decimal.Decimal(str(1251.0000014576121234854513))) == decimal.Decimal(
         str(1251.00000145))
+    assert personal_data.decimal_adapt_price(symbol_market,
+                                             decimal.Decimal(str(1251.0000014576121234854513)),
+                                             True) == decimal.Decimal(
+        str(1251.00000145))
```
Comment (Herklos) [inline tests/personal_data/orders/test_decimal_order_adapter.py:63]: :+1:

File: tests/personal_data/orders/test_decimal_order_adapter.py [hunk 2]
```diff
@@ -404,11 +428,16 @@ async def test_adapt_quantity():
...
+    assert personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(1.00000000001), 10) == decimal.Decimal(1)
+    assert personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal("1.01"), 1, True) == decimal.Decimal(1)
+    assert personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal("1.01"), 1, False) \
+        == decimal.Decimal("1.1")
     assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(1.00000000001), 10)) == 1
     assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(1.00000000001), 11)) == 1.00000000001
     assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(578.000145000156), 3)) == 578
     assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(578.000145000156), 4)) == 578.0001
     assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(578.000145000156), 7)) == 578.000145
+    assert float(personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(578.000145000156), 7)) == 578.000145
```
Comment (Herklos) [inline tests/personal_data/orders/test_decimal_order_adapter.py:440]: Is there any difference between this two lines ?

File: tests/personal_data/orders/test_decimal_order_adapter.py [hunk 3]
```diff
@@ -404,11 +428,16 @@ async def test_adapt_quantity():
 
 
 async def test_decimal_trunc_with_n_decimal_digits():
+    assert personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal(1.00000000001), 10) == decimal.Decimal(1)
+    assert personal_data.decimal_trunc_with_n_decimal_digits(decimal.Decimal("1.01"), 1, True) == decimal.Decimal(1)
```
Comment (Herklos) [inline tests/personal_data/orders/test_decimal_order_adapter.py:432]: :+1:

File: octobot_trading/personal_data/orders/groups/balanced_take_profit_and_stop_order_group.py [hunk 4]
```diff
@@ -72,6 +72,19 @@ def can_create_order(self, order_type, quantity):
             return balance[self.STOP].get_balance() + quantity <= balance[self.TAKE_PROFIT].get_balance()
         return balance[self.TAKE_PROFIT].get_balance() + quantity <= balance[self.STOP].get_balance()
 
+    def get_max_order_quantity(self, order_type):
+        """
+        Returns The maximum order quantity to reach the order side's balance
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/groups/balanced_take_profit_and_stop_order_group.py:77]: :+1:

File: tests/personal_data/orders/test_decimal_order_adapter.py (hunk 2 above)
Comment (GuillaumeDSM) [reply tests/personal_data/orders/test_decimal_order_adapter.py]: updated

**Review by Herklos — APPROVED** [review]

**Pull request merged**